packages/tailwindcss/src/ast.ts

```diff
@@ -297,7 +301,15 @@ export function optimizeAst(ast: AstNode[]) {
       for (let child of node.nodes) {
         transform(child, copy.nodes, depth + 1)
       }
-      parent.push(copy)
+      if (
+        copy.nodes.length > 0 ||
+        copy.name === '@layer' ||
+        copy.name === '@charset' ||
+        copy.name === '@custom-media' ||
+        copy.name === '@namespace'
```

Comment:
Should this just be all at-rules? 🤔

Reply:
```cs
@media (foo) {}
```
Should be removed imo

Reply:
yeah i noticed this too, you just responded before i could update my comment heh

Reply:
Hehe, I think in theory we could invert the condition and don't touch everything but the `@media` (but then we need to check for `@suspports` as well. 

In a perfect world `@layer foo {}` is also removed, (because you should use `@layer foo;` but we don't make that distinction.